src/librustc_typeck/coherence/mod.rs

```diff
@@ -258,9 +258,8 @@ impl<'a, 'gcx, 'tcx> CoherenceChecker<'a, 'gcx, 'tcx> {
                                                    type `{}`",
                                                    self_type);
                     err.span_label(span,
-                                   &format!("field `{}` in struct `{}` doesn't implement `Copy`",
-                                            name,
-                                            self_type));
+                                   &format!("field `{}` doesn't implement `Copy`",
+                                            name));
                     if let Some(def_span) = tcx.map.span_if_local(did) {
                         err.span_label(def_span, &"this field doesn't implement `Copy`");
                     }
```

Comment:
@jonathandturner We should talk about a transition plan: ideally, `span_if_local` wouldn't exist after #37954, as every `DefId` has an associated `Span`.
The problem is that there's no source available and there's a few ways in which we could approach this (including embedding the full source in metadata).
In any case, long-term I'd like to not have code that branches out like this, but rather generate "field bar of struct Foo" from just its `DefId`, and point to its span if local, *unless* we can somehow show the source itself.